packages/zoe/README.md

```diff
@@ -124,3 +132,25 @@ harden(prepare);
 ```
 
 For an example contract upgrade, see the test at https://github.com/Agoric/agoric-sdk/blob/master/packages/zoe/test/swingsetTests/upgradeCoveredCall/test-coveredCall-service-upgrade.js .
+
+### Durability
+
+The contract must retain in durable storage anything that must persist between incarnations. All other state will be lost.
+
+### Kinds
+
+The contract defines the kinds that are held in durable storage. Thus the function calls that define the kinds must be run before the objects are deserialized from durable storage.
```

Comment:
The "Kind" apis defineDurableKind, etc, are deprecated, and I hope to see all of them replaced by the Exo instance and ExoClass apis soon. The concept of "Kind" does still appear in the KindHandle if you're using the low level durable api directly. However, most code will hopefully be using prepare*. For that code, there is no exposed notion of "Kind". Just Exo instances and ExoClasses.

Reply:
I'm not sure how to best document this relationship. Swingset (via the "liveslots" layer) provides `globalThis.VatData`, which provides `defineKind` and friends, `makeKindHandle`, `makeScalarBigMapStore` and friends, and a few others for "watched promises". These are [documented](https://github.com/Agoric/agoric-sdk/blob/c2c0da9dc1e49b84b3ede332a70c409a393ab6e7/packages/SwingSet/docs/virtual-objects.md#virtual-and-durable-objects) by SwingSet. 

https://github.com/Agoric/agoric-sdk/blob/c2c0da9dc1e49b84b3ede332a70c409a393ab6e7/packages/swingset-liveslots/src/liveslots.js#L1316-L1328

The `@agoric/vat-data` package wraps these with additional layers, like Exo stuff (although I don't see any prose documentation [in that directory](https://github.com/Agoric/agoric-sdk/tree/c2c0da9dc1e49b84b3ede332a70c409a393ab6e7/packages/vat-data) .. do we document them anywhere?).

I'm happy for our docs to recommend Exo/`prepare`/etc as the preferred interface, but the fundamental Swingset requirement is about `makeKindHandle` and `defineDurableKind`, and the error messages that misbehaving users/code will get are going to cite those functions. So I think somewhere in the docs, we'll need to provide a reference to the underlying facilities that `Exo`/`prepare` wrap, so folks can search for the swingset error message and find directions to the recommended wrapper functions they must call.